Stratis.Bitcoin/BlockPulling/LookaheadBlockPuller.cs

```diff
@@ -397,7 +375,13 @@ private Block NextBlockCore(CancellationToken cancellationToken)
                 cancellationToken.ThrowIfCancellationRequested();
                 ChainedBlock header = this.Chain.GetBlock(this.location.Height + 1);
                 DownloadedBlock block;
-                if (header != null && TryRemoveDownloadedBlock(header.HashBlock, out block))
+
+                bool isDownloading = false;
+                bool isReady = false;
+                if (header != null) CheckBlockStatus(header.HashBlock, out isDownloading, out isReady);
```

Comment:
This logic will make my tuple suggestion hard to implement so using "out" looks fine.